From 8ed3a23d233da4602b7c649139d3ba38abb0156c Mon Sep 17 00:00:00 2001
From: Artem Polishchuk <ego.cordatus@gmail.com>
Date: Sat, 14 Nov 2020 23:19:44 +0200
Subject: [PATCH 1/3] feat: add AppData manifest
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Every software center that exists allows the user to look at
screenshots and a long description of the application before it is
installed. For most users it allows them to answer the question “Do I
want to install this application?”. Traditionally in Linux
distributions, we have none of this data for the vast majority of our
desktop user-installable applications.

More info:
https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html
---
 .../linux/data/nz.mega.MEGAsync.metainfo.xml  | 88 +++++++++++++++++++
 1 file changed, 88 insertions(+)
 create mode 100644 src/MEGASync/platform/linux/data/nz.mega.MEGAsync.metainfo.xml

diff --git a/src/MEGASync/platform/linux/data/nz.mega.MEGAsync.metainfo.xml b/src/MEGASync/platform/linux/data/nz.mega.MEGAsync.metainfo.xml
new file mode 100644
index 00000000..9354165b
--- /dev/null
+++ b/src/MEGASync/platform/linux/data/nz.mega.MEGAsync.metainfo.xml
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2020 Artem Polishchuk <ego.cordatus@gmail.com> -->
+<component type="desktop-application">
+  <id>nz.mega.MEGAsync</id>
+  <metadata_license>CC0-1.0</metadata_license>
+
+  <!-- MEGAsync is under a proprietary license, except the SDK which is BSD -->
+  <project_license>LicenseRef-proprietary</project_license>
+
+  <launchable type="desktop-id">megasync.desktop</launchable>
+  <provides>
+    <binary>megasync</binary>
+  </provides>
+
+  <content_rating type="oars-1.0">
+  </content_rating>
+
+  <summary>
+    Easy automated syncing between your computers and your MEGA Cloud Drive.
+  </summary>
+
+  <description>
+    <p>Easy automated synchronisation between your computer and your MEGA
+    cloud.</p>
+
+    <p>Sync and lots more with the MEGA Desktop App</p>
+
+    <p>For all major platforms — Compatible with Windows, macOS and Linux,
+    including syncing between different platforms.</p>
+
+    <p>Stream files directly from MEGA — The MEGA app allows you to stream any
+    file from your MEGA cloud or from a file link, straight to your favourite
+    player.</p>
+
+    <p>Deleted data retention — MEGA moves deleted files to a special folder on
+    your local computer and in your MEGA cloud so they can be restored if you
+    need to.</p>
+
+    <p>Powerful Transfer Manager — Full control over active and completed
+    transfers: prioritise, pause/resume, open, generate links and more…</p>
+
+    <p>Easy Automated Synchronisation — Access and work with your data securely
+    across different locations and devices. It’s always on and fully
+    automatic.</p>
+
+    <p>Browser Integration — All MEGA browser file transfers can be
+    automatically handled by the MEGA app, significantly enhancing the
+    reliability, performance and speed. This avoids browser limitations,
+    especially for large files.</p>
+
+    <p>File Manager — As well as syncing specified folders, you can upload and
+    download selected files/folders, enable file versioning and overlay sync
+    icons in your standard file explorer.</p>
+  </description>
+
+  <releases>
+    <release date="@DATE@" version="@VERSION@"></release>
+    <release date="2020-07-27" version="4.3.3.0"></release>
+  </releases>
+
+  <screenshots>
+    <screenshot type="default">
+      <image>https://eu.static.mega.co.nz/3//images/mega/bottom-page/sp_sync_1_v3@2x.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://eu.static.mega.co.nz/3//images/mega/bottom-page/sp_sync_2_v2@2x.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://eu.static.mega.co.nz/3//images/mega/bottom-page/sp_sync_3_v2@2x.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://eu.static.mega.co.nz/3//images/mega/bottom-page/sp_sync_4@2x.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://eu.static.mega.co.nz/3//images/mega/bottom-page/sp_sync_5@2x.png</image>
+    </screenshot>
+  </screenshots>
+
+  <keywords>
+    <keyword>cloud</keyword>
+    <keyword>filesharing</keyword>
+    <keyword>sync</keyword>
+  </keywords>
+
+  <url type="bugtracker">https://github.com/meganz/MEGAsync/issues</url>
+  <url type="help">https://mega.nz/help/client/megasync/</url>
+  <url type="homepage">https://mega.nz/sync</url>
+</component>

From 8fed40c06cfe917f25fb583a239e32580cb4878c Mon Sep 17 00:00:00 2001
From: Artem Polishchuk <ego.cordatus@gmail.com>
Date: Thu, 19 Nov 2020 17:46:59 +0200
Subject: [PATCH 2/3] build: install AppData manifest

---
 src/MEGASync/platform/platform.pri | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/MEGASync/platform/platform.pri b/src/MEGASync/platform/platform.pri
index 1864f41e..d26b54f1 100644
--- a/src/MEGASync/platform/platform.pri
+++ b/src/MEGASync/platform/platform.pri
@@ -68,6 +68,11 @@ unix:!macx {
         desktop.commands = update-desktop-database &> /dev/null || true
         INSTALLS += desktop
 
+        # metainfo
+        metainfo.path = $$DESKTOP_DESTDIR/share/metainfo
+        metainfo.files = $$PWD/linux/data/nz.mega.MEGAsync.metainfo.xml
+        INSTALLS += metainfo
+
         HICOLOR = $$DESKTOP_DESTDIR/share/icons/hicolor
 
         # icons

From 95b7a393767dc74d30d4c07e6571ba616bff4b87 Mon Sep 17 00:00:00 2001
From: Artem Polishchuk <ego.cordatus@gmail.com>
Date: Mon, 23 Nov 2020 16:26:38 +0200
Subject: [PATCH 3/3] fix: add name tag into AppData

Mandatory for Flathub AppData validation tool.
---
 src/MEGASync/platform/linux/data/nz.mega.MEGAsync.metainfo.xml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/MEGASync/platform/linux/data/nz.mega.MEGAsync.metainfo.xml b/src/MEGASync/platform/linux/data/nz.mega.MEGAsync.metainfo.xml
index 9354165b..b8ff377b 100644
--- a/src/MEGASync/platform/linux/data/nz.mega.MEGAsync.metainfo.xml
+++ b/src/MEGASync/platform/linux/data/nz.mega.MEGAsync.metainfo.xml
@@ -7,6 +7,7 @@
   <!-- MEGAsync is under a proprietary license, except the SDK which is BSD -->
   <project_license>LicenseRef-proprietary</project_license>
 
+  <name>MEGAsync</name>
   <launchable type="desktop-id">megasync.desktop</launchable>
   <provides>
     <binary>megasync</binary>
